src/sherpa_ai/memory/shared_memory_with_vectordb.py

```diff
@@ -0,0 +1,64 @@
+from typing import List, Optional
+
+from langchain.embeddings.openai import OpenAIEmbeddings
+
+from sherpa_ai.actions.planning import Plan
+from sherpa_ai.agents import AgentPool
+from sherpa_ai.connectors.chroma_vector_store import ChromaVectorStore
+from sherpa_ai.events import Event, EventType
+from sherpa_ai.memory import SharedMemory
+from sherpa_ai.memory.belief import Belief
+
+
+class SharedMemoryWithVectorDB(SharedMemory):
+    """
+    Custom implementation of SharedMemory that integrates with ChromaVectorStore.
+
+    Use this class whenever context retrieval from a vector database is needed.
+
+    Attributes:
+        session_id (str): Unique identifier for the current session.
+
+    """
+
+    def __init__(
+        self,
+        objective: str,
+        session_id: str,
+        agent_pool: AgentPool = None,
+    ):
+        self.objective = objective
+        self.agent_pool = agent_pool
+        self.events: List[Event] = []
+        self.plan: Optional[Plan] = None
+        self.current_step = None
+        self.session_id = session_id
+
+    def observe(self, belief: Belief):
+        vec_db = ChromaVectorStore.chroma_from_existing()
```

Comment:
Currently, this is only for Chromadb. Is there another PR that will address the idea of being able to select the vectordb backend?

Reply:
Also, this should be in the constructor `__init__` since we only need to get the vectordb once I think

Reply:
we have a pr to handle multiple vector db's

Reply:
Then please put this to the __init__ function. We don't need to connect to the vector database everytime we make an observation